src/storage/model.cpp

```diff
@@ -90,6 +89,26 @@ storm::storage::BitVector statesWithParameter(SparseModel<RationalFunction> cons
     return result;
 }
 
+template<typename ValueType>
+bool hasCycle(SparseModel<ValueType> const& model) {
+    // Find all states with absorbing self-loops and exclude them
+    auto const& matrix = model.getTransitionMatrix();
+    storm::storage::BitVector subStates(matrix.getColumnCount(), true);
+    for (auto group = 0; group < matrix.getRowGroupCount(); ++group) {
+        for (auto& entry : matrix.getRowGroup(group)) {
+            if (entry.getColumn() == group) {
+                if (storm::utility::isOne<ValueType>(entry.getValue())) {
+                    // Found self-loop
+                    subStates.set(entry.getColumn(), false);
+                }
+            }
+            if (entry.getColumn() >= group) {
+                continue;
+            }
+        }
+    }
+    return storm::utility::graph::hasCycle(matrix, subStates);
```

Comment:
This is problematic for MDPs, when  we have one self-loop row first and then one non-selfloop row which induces a cycle with other states, right?

I think the "hasCycle" would need to have a "subChoices" input to make this work?
Alternatively, this method could also build an SCC decomposition and we then check if each SCC has size 1.

Reply:
You are right, this is indeed problematic.
I am open for other suggestion. The SCC decomposition might be a good idea.
I actually just wanted to have a simple function to check for acyclity. Unfortunately, `hasCycle` is not exactly giving me what I want. Maybe it makes sense to implement a better function in Storm directly.